CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+project(igrec)
+cmake_minimum_required(VERSION 2.8)
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "RelWithAsserts" CACHE STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
+endif(NOT CMAKE_BUILD_TYPE)
+
+add_subdirectory(configs)
+add_subdirectory(src)
+
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX "/usr/local" CACHE PATH "default install path" FORCE)
```

Comment:
1) Linux: We should decide either follow standard Linux way of` make install` with `sudo` or not.

2) Mac OS X: At least at El Capitan 10.11.4 I have full access to `/usr/local` (AFAIK is it due my usage of Homebrew). Anyway on Mac OS `/usr/local/` is not considered to be "system folder", so installing user stuff there is perfectly fine. However, I would suggest installing by package manager: either Homebrew (normally without sudo) or MacPorts (normally with sudo).